packages/studio/src/store/StudioActions.ts

```diff
@@ -183,6 +185,26 @@ export default class StudioActions {
     this.getPages().setActivePageEntities(entitiesRecord);
   };
 
+  addLayout = (layoutState: LayoutState) => {
+    const tree = this.getComponentTree() ?? [];
+    const layoutTreeWithNewUUIDs: ComponentState[] =
+      ComponentTreeHelpers.mapComponentTreeParentsFirst(
+        layoutState.componentTree,
+        (componentState, parentState) => {
+          const updatedComponentState = {
+            ...componentState,
+            uuid: v4(),
+          };
+          if (parentState) {
+            updatedComponentState.parentUUID = parentState.uuid;
+          }
+          return updatedComponentState;
+        }
+      );
+    const updatedTree = layoutTreeWithNewUUIDs.concat(tree);
```

Comment:
I recall overhearing at some point that we were planning on placing the new layout at the end of the component tree-- did product decide on adding it to the beginning?

Reply:
Aaron says top!